api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -165,9 +166,25 @@ message FilterChainMatch {
   repeated string application_protocols = 10;
 }
 
+// The configuration for on-demand filter chain. If this field is not empty in FilterChain message,
+// a filter chain will be built on-demand.
+// On-demand filter chains will be first built as placeholders when a listener is created in order to
+// speedup the warming up of the listener. The building and initialization of an on-demand filter chain
+// will be postponed to the arrival of new connections that require this filter chain.
+// The building process of on-demand filter chains will be similar to the one of normal filter chains.
+// However, a unique rebuilder with a unique init manager will be assigned to build each on-demand filter chain.
+// On the completion of building the on-demand filter chain, connections that require this filter chain will be retried.
+message OnDemandConfiguration {
+  // [#not-implemented-hide:] The timeout to wait for filter chain placeholders to complete rebuilding.
+  // If the timeout is reached, the filter chain rebuilding fails. Rebuilder send callbacks with failure signal
+  // so that end users will close all stored connections related to this filter chain.
+  // Specify 0 to disable the timeout. If not specified, a default timeout of 15s is used.
+  google.protobuf.Duration rebuild_timeout = 1;
+}
```

Comment:
Make this a nested message of FilterChain

Reply:
Good idea! Updated.